clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -92,7 +100,7 @@ class BuiltinNameEmitter {
   // whether a function is a builtin function.
   void Emit();
 
-private:
+protected:
```

Comment:
this class is declared in a cpp file and in an annonymous namespace, so the impact of this change is extremely limited. This is to avoid code duplication and separate the 2 intents of the classes.

Ideally, there should be a based class handling the processing of the td file and 2 emitters (as there is no logical relationship between a name emitter and JSON). But this involve an extra burden to maintain this in sync with upstream.